pykube/oidc.py

```diff
@@ -0,0 +1,89 @@
+"""
+Open ID Connect (OIDC) related code.
+"""
+
+import base64
+import json
+import requests
+import time
+
+
+# If our token is about to expire we should refresh in anticipation
+expiry_buffer = 10
+
+
+def _pad_b64(b64):
```

Comment:
The token format is of a JSON Web Token, which is a dot separated series of base64url-encoded values ([RFC 7519](https://tools.ietf.org/html/rfc7519#section-3)). base64url-encoded values may omit the buffer ([RFC 4648](https://tools.ietf.org/html/rfc4648#section-5)):

> The pad character "=" is typically percent-encoded when used in an URI, but if the data length is known implicitly, this can be avoided by skipping the padding; see section 3.2.

In practice, the OIDC provider I've been using, Google, provides token IDs that are not padded.